pkg/utilization/utilisation.go

```diff
@@ -137,6 +139,17 @@ type ExecutionConfig struct {
 // If Config.Skip is set to true, the provided function will be run without recording metrics.
 // If there is an error with initiating the metrics recorder, the function will be run without recording metrics.
 func WithMetricsRecorder(config Config, fn func(), postProcessFn func() error) {
+	var err error
+	defer func() {
+		if err != nil {
```

Comment:
in `if err := postProcessFn(); err != nil {` it will be local err

Reply:
nice catch, thanks!